File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
        String.strip(input) == "" -> "Fine. Be that way!"
        question?(input) -> "Sure."
        shouting?(input) -> "Whoa, chill out!"
        true -> "Whatever."
    end
  end

  def shouting? (message) do
    message == String.upcase(message)
    && (String.downcase(message) != message)
  end
  def question? (message) do
    String.ends_with?(message, "?")
  end
end

File: python/bob/bob.py
#
# Skeleton file for the Python "Bob" exercise.
#

import re

def hey(what):
  what = re.sub("[\s\t]", "", what)

  if what == "":
    return "Fine. Be that way!"
  elif what.upper() == what and what.lower() != what :
    return "Whoa, chill out!"
  elif what[-1] == "?":
    return "Sure."
  return "Whatever."
